astro/src/content/docs/customize/look-and-feel/advanced-theme-editor.mdx

```diff
@@ -0,0 +1,184 @@
+  </APIField>
+  <APIField name="Name" required>
+    A unique name to identify the theme. This name is for display purposes only and it can be modified later if desired.
+  </APIField>
+</APIBlock>
+
+
+### Templates
+
+{/* ===== */}
+{/*  To add a new theme template, do the following */}
+{/*  update site/_date/templates.yaml (further instructions there) */}
+{/*  update the JSON files in site/docs/src/json/themes/ with the new theme template key */}
+{/*  touch this file to regenerate (if in dev mode) */}
+{/*  that's it. the API and the theme form page will be automatically updated. */}
+
+<APIBlock>
+  <APIField name="Stylesheet (CSS)" optional>
+    This CSS stylesheet may be used to style the themed pages.
+
+    This CSS will be included in the `head` tag in the Helpers `head` macro. You may also choose to include other remote stylesheets by using the `<style>` tag within the `head` macro.
+
+    ```
+    <style>
+      ${theme.stylesheet()}
+    </style>
+    ```
+  </APIField>
+  <APIField name="Messages" optional>
+    This section allows you to add additional localized messages to your theme. When creating an additional locale it is not required that all messages are defined for each language. If a message key is not defined for the specified locale, the value from the default bundles will be used.
+
+    If you intend to localize your login templates, you may find our [community contributed and maintained messages in our GitHub repository](https://github.com/FusionAuth/fusionauth-localization) helpful.
+  </APIField>
+  <APIField name="Helpers" required>
+    This template contains all of the main helper macros to define the `head`, `body` and `footer`. To begin theming FusionAuth you'll want to start with this template as it will affect all other templates.
+
+    See the [Helpers](/docs/customize/look-and-feel/helpers) page for additional information.
+  </APIField>
+</APIBlock>
+
+<Templates />
+
+## Preview a Theme
+
+If you want to see how your theme works, you can always open a browser with no active FusionAuth session and visit the hosted login pages.
+
+However, at times, you may need to make changes in your theme that you want to view without going through an entire registration process. You can easily do so by previewing the theme via the administrative user interface.
+
+Navigate to <strong>Customizations -> Themes</strong>. Choose your theme, then click the preview link (the green magnifying glass):
+
+<img src="/img/docs/customize/look-and-feel/theme-preview-button.png" alt="Preview your theme" width="1200" role="bottom-cropped" />
+
+This will open a new tab. Click on any of the pages you've modified in the left hand navigation, for example <InlineField>OAuth register</InlineField>, and you'll see the page as it would be rendered.
+
+## Example Code
+
+### Displaying Messages
+
+You can customize messages by locale. You can also have optional keys.
+
+{/*
+<MessagesExample />
```

Comment:
should this be commented out?

Reply:
Some joker did that in Oct of last year :) .

```
8eefac3917 (Lyle Schemmerling 2023-10-31 11:19:09 -0600 136) {/*
8eefac3917 (Lyle Schemmerling 2023-10-31 11:19:09 -0600 137) <MessagesExample />
8eefac3917 (Lyle Schemmerling 2023-10-31 11:19:09 -0600 138) */}
```

Will take a look.

Reply:
Looks fine to me.